pkg/apis/core/v1beta1/validation/validation.go

```diff
@@ -17,30 +17,117 @@ limitations under the License.
 package validation
 
 import (
+	"strings"
+
+	apiextv1b1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apimachineryval "k8s.io/apimachinery/pkg/api/validation"
+	valutil "k8s.io/apimachinery/pkg/util/validation"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 
 	"sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
 )
 
-func ValidateFederatedTypeConfig(object *v1beta1.FederatedTypeConfig) field.ErrorList {
+func ValidateFederatedTypeConfig(obj *v1beta1.FederatedTypeConfig, statusSubResource bool) field.ErrorList {
+
+	allErrs := ValidateFederatedTypeConfigSpec(&obj.Spec, field.NewPath("spec"))
+	if statusSubResource {
+		allErrs = append(allErrs, ValidateFederatedTypeConfigStatus(&obj.Status, field.NewPath("status"))...)
+	}
+	return allErrs
+}
+
+func ValidateFederatedTypeConfigSpec(spec *v1beta1.FederatedTypeConfigSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := ValidateAPIResource(&spec.TargetType, fldPath.Child("targetType"))
```

Comment:
Why does validation of the target type differ from validation of the federated and status types in more than how the group is evaluated?